File: py_pubsub/py_pubsub/string_publisher.py
#!/usr/bin/env python3
import rclpy
from rclpy.node import Node
from example_interfaces.msg import String
import rclpy.publisher


class StringPublisher(Node): 
    def __init__(self):
        super().__init__("string_publisher") 
        self.publisher = self.create_publisher(String,"my_message",qos_profile=10)
        self.timer = self.create_timer(1,self.stringCB)
        self.get_logger().info("The Node started!")

    def stringCB(self):
        msg = String()
        msg.data = "I'am a String publisher!"
        self.publisher.publish(msg)


def main(args=None):
    rclpy.init(args=args)
    node = StringPublisher()
    rclpy.spin(node)
    rclpy.shutdown()


if __name__ == "__main__":
    main()

File: cpp_pubsub/src/string_publisher.cpp
#include "rclcpp/rclcpp.hpp"
#include "example_interfaces/msg/string.hpp"

class StringPublisher : public rclcpp::Node 
{
public:
    StringPublisher() : Node("string_publisher") 
    {
        publisher = this->create_publisher<example_interfaces::msg::String>("string_msg", 10);
        timer = this->create_wall_timer(std::chrono::milliseconds(500),
                                         std::bind(&StringPublisher::publishStringCB, this));
        RCLCPP_INFO(this->get_logger(), "String publisher started!");
    }

private:

    void publishStringCB()
    {
        auto msg = example_interfaces::msg::String();
        msg.data = std::string("Hi Im from string publisher node!");
        publisher->publish(msg);
    }

    rclcpp::Publisher<example_interfaces::msg::String>::SharedPtr publisher;
    rclcpp::TimerBase::SharedPtr timer;


};

int main(int argc, char **argv)
{
    rclcpp::init(argc, argv);
    auto node = std::make_shared<StringPublisher>();
    rclcpp::spin(node);
    rclcpp::shutdown();
    return 0;
}
